File: 237-delete-node-in-a-linked-list/delete-node-in-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    void deleteNode(ListNode* node) {
        ListNode* curr=node;
         ListNode*nextt=node->next;
          ListNode* prev=nullptr;
        curr=node;

        curr->val =nextt->val;
        curr->next = nextt->next;

        
    }
};


// 1 -> 2-> 3 -> nullptr 

File: 237-delete-node-in-a-linked-list/delete-node-in-a-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

/**
 Do not return anything, modify it in-place instead.
 */
function deleteNode(node: ListNode | null): void {
    let prev:(ListNode | null)=node;
    let temp:(ListNode | null)=null;
    node=node.next;
    while(node!=null){
        prev.val=node.val;
        temp=prev;
        prev=node;

        node=node.next;
        // 
    }
    temp.next=null;
};